app/models/project.rb

```diff
@@ -704,9 +722,38 @@ def check_status
       update(status: nil, audit_comment: "Response #{response.response_code}")
     end
     # only update status to nil if the response code is a success
-  rescue RateLimitService::OverLimitError => e
-    StructuredLog.capture("CHECK_STATUS_RATE_LIMITED", { platform: platform, name: name, exceeded_by: e.exceeded_by })
-    raise CheckStatusInternallyRateLimited
+  end
+
+  # Special-case of fetching package url for NPM, where we observe their rate-limiting and raise an error
+  # if we're exceeding their "retry-after" value.
+  def rate_limited_npm_request(url)
+    # This key needs to be host-specific so hosts don't overwrite each other's different values.
+    key = "#{Project::CHECK_STATUS_NPM_RETRY_AFTER}:#{Socket.gethostname}"
```

Comment:
fetching the hostname here is like 20 microseconds, so we probably don't need to worry about caching it in the process.